packages/core/src/model.ts

```diff
@@ -80,23 +110,122 @@ export class ChartModel {
 	 */
 	setData(newData) {
 		const sanitizedData = this.sanitize(Tools.clone(newData));
-		const dataLabels = this.generateDataLabels(sanitizedData);
+		const dataGroups = this.generateDataGroups(sanitizedData);
 
 		this.set({
 			data: sanitizedData,
-			dataLabels
+			dataGroups
 		});
 
 		return sanitizedData;
 	}
 
-	generateDataLabels(newData) {
-		const dataLabels = {};
-		newData.datasets.forEach(dataset => {
-			dataLabels[dataset.label] = Configuration.legend.items.status.ACTIVE;
+	/*
+	 * Data groups
+	*/
+	protected updateAllDataGroups() {
+		// If allDataGroups hasn't been initialized yet
+		// Set it to the current set of data groups
+		if (!this.allDataGroups) {
+			this.allDataGroups = this.getDataGroups().map(group => group.name);
+		} else {
+			// Loop through current data groups
+			this.getDataGroups().forEach(dataGroup => {
```

Comment:
`allDataGroups` is used to generate a color scale that applies to all the groups.

Now when the data updates, you might remove a group, and then bring it back in a newer date update, therefore the order of the groups in `allDataGroups` matters so that you'd never have an incorrect color assigned to it.

Also, a new `group` going into `allDataGroups` shouldn't have existed in there before, hence why the `forEach` and the `if` inside of that.

Reply:
That makes sense. Maybe add a comment that explains that?